### PR TITLE
Adds arctangent 2 component to circuits

### PR DESCRIPTION
**Pull request by Time-Green**
## About The Pull Request
Adds an arctangent 2 component (atan2)

## Why It's Good For The Game

It's made to convert offsets into angles, something thats pretty useful in a coordinate based game
## Changelog
:cl:
add: Adds an arctangent2 component to circuitry!
/:cl:

### CODE REVIEW

**Review by Watermelon914 — APPROVED** [review]

**Comment by DATA-xPUNGED** [comment]
circuit powercreep... millions must code

**Comment by MTandi** [comment]
Is this for turrets?

**Comment by Time-Green** [comment]
> Is this for turrets?

I would never use circuits in such an evil way

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -384,6 +384,7 @@
 		"comp_tonumber",
 		"comp_tostring",
 		"comp_trigonometry",
+		"comp_arctan2",
```
Comment (Mothblocks) [inline code/modules/research/techweb/all_nodes.dm:387]: ABC order

File: code/modules/wiremod/components/math/arctan2.dm [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	var/datum/port/input/input_port_y
+
+	/// The result from the output
+	var/datum/port/output/output
+
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+
+/obj/item/circuit_component/arctan2/populate_ports()
+	input_port_x = add_input_port("Input", PORT_TYPE_NUMBER)
+	input_port_y = add_input_port("Input", PORT_TYPE_NUMBER)
```
Comment (Mothblocks) [inline code/modules/wiremod/components/math/arctan2.dm:22]: Should these not say "X" and "Y"?

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/wiremod/components/math/arctan2.dm (hunk 2 above)
Comment (Time-Green) [reply code/modules/wiremod/components/math/arctan2.dm:22]: Do they not or am I tripping?

Comment (Mothblocks) [reply code/modules/wiremod/components/math/arctan2.dm:22]: I'm referring to the names--this is just "Input" and "Input". Shouldn't they be "X" and "Y"?

Comment (Time-Green) [reply code/modules/wiremod/components/math/arctan2.dm:22]: Oh, maybe?

**Pull request merged**